src/solvers/NelderMead.jl

```diff
@@ -280,22 +284,45 @@ function step_solver!(mp::AbstractManoptProblem, s::NelderMeadState, ::Any)
         for i in 2:length(ind)
             retract!(
                 M,
-                s.population.pts[ind[i]],
-                s.population.pts[ind[1]],
+                s.population.pts[i],
+                s.population.pts[1],
                 inverse_retract(
-                    M,
-                    s.population.pts[ind[1]],
-                    s.population.pts[ind[i]],
-                    s.inverse_retraction_method,
+                    M, s.population.pts[1], s.population.pts[i], s.inverse_retraction_method
                 ),
                 s.σ,
                 s.retraction_method,
             )
             # update cost
-            s.costs[ind[i]] = get_cost(mp, s.population.pts[ind[i]])
+            s.costs[i] = get_cost(mp, s.population.pts[i])
         end
     end
     # store best
     s.p = s.population.pts[argmin(s.costs)]
     return s
 end
+
+mutable struct NelderMeadPopulationCriterion{TF<:Real,TX<:Real} <: StoppingCriterion
+    tol_f::TF
+    tol_x::TX
+    reason::String
+    function NelderMeadPopulationCriterion(tol_f::Real=1e-8, tol_x::Real=1e-8)
+        return new{typeof(tol_f),typeof(tol_x)}(tol_f, tol_x, "")
+    end
+end
+
+function (c::NelderMeadPopulationCriterion)(
```

Comment:
I usually prefer names with “StopWhen” and it seems to be a “If the costs differ less enough from the first and the distances differ less enough”?

Maybe `StopWhenPopulationConcentrated` or something?

Reply:
OK, I'll rename it to `StopWhenPopulationConcentrated`.